### PR TITLE
fix: Rescheduling email when there is broken calendar integration

### LINKED ISSUE #15439: [CAL-3937] Confirmation email is not being send out when adding more attendees.
If we try to add more attendees 
upcoming bookings > edit> reschedule > add participants. it allows you to add another participant to the meeting but does not send a confirmation email to the new attendee.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3937](https://linear.app/calcom/issue/CAL-3937/confirmation-email-is-not-being-send-out-when-adding-more-attendees)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- Fixes  #15439 
- Fixes CAL-3937

The rescheduling emails were not sent in case of a failure with one of the apps/integrations that could be a Calendar App or a conferencing app. This was intentional because a separate Broken Integration Email goes in that case that has the booking details as well as the information on which App is broken. The email has a clear subject inviting the user to take action on the issue.
image

## There are two reasons why it was a problem:
- Broken integration email was sent only when a Conferencing App integration fails. For a calendar integration, there was no Broken integration email being sent. This resulted in a scenario where the meeting is successfully rescheduled, calendars are updated, webhooks are sent but no reschedule email is sent to any of the organizer or attendee.
- Broken Integration Email is sent only to the organizer and thus no other attendee has email communication about the reschedule.

## New Approach
Send rescheduled email as well as Broken Integration Email(not fixing the Calendar App Broken integration email in this PR)

I think it is perfectly fine for the organizer to receive two emails in that case. One informing about the reschedule and the other informing about the broken App.

Broken Integration Email
image

Reschedule Email
image

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
- Book an event that has CalVideo location set.
- Remove Daily Video env variables so that Video creation fails
- Reschedule the booking and notice that Broken Integration Email along with the regular reschedule emails will go.

### CODE REVIEW

**Comment by hariombalhara** [comment]
* **#15669** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/15669?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hariombalhara and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -1799,8 +1799,7 @@ async function handler(
 
     evt.appsStatus = handleAppsStatus(results, booking, reqAppsStatus);
 
-    // If there is an integration error, we don't send successful rescheduling email, instead broken integration email should be sent that are handled by either CalendarManager or videoClient
-    if (noEmail !== true && isConfirmedByDefault && !isThereAnIntegrationError) {
```
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking.ts:1803]: I did not receive any broken integration mail when google calendar integration fails

Comment (Udit-takkar) [reply packages/features/bookings/lib/handleNewBooking.ts:1803]: The event was not updated as credentials don't work and i got successfull reschedule email

Comment (hariombalhara) [reply packages/features/bookings/lib/handleNewBooking.ts:1803]: Yeah, that isn't fixed in this PR. That's a separate task pending from a long time now 
image

**Review by Udit-takkar — COMMENTED** [review]
Left few comments

**Review by Udit-takkar — APPROVED** [review]

**Review by Udit-takkar — DISMISSED** [review]
Need to figure out few things before we merge this

**Comment by hariombalhara** [comment]
Will discuss it more, but here is my opinion on this PR.

The question is how does one consider the rescheduling has been done.

- Booking is rescheduled in our DB first. If that fails, then booking fails and user is notified about it during booking.

- If Calendar event update fails, people haven't been notified through calendar about it.

- We weren't sending the rescheduled emails earlier, so they weren't notified about it through email

- But the webhook fires. So, as per webhook the rescheduling occured.

So, if we go with the approach of failing the booking itself if calendar update fails, then the inconsistency won't arise. But that's a big decision and a separate discussion.

I think doing the change in the PR atleast improves the situation as DB, email communication and webhooks are in sync as these are the one we control directly.

**Comment by hariombalhara** [comment]
Discussed internally, we are good to go @Udit-takkar !! Thanks for stopping me and suggesting to discuss it 🙏

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**